docs/developer/stock-allocation.mdx

```diff
@@ -0,0 +1,298 @@
+---
+title: Stock allocation
+---
+
+## Introduction
+
+This guide explains how the allocations are handled in Saleor.
+
+## Terms introduction
+
+The warehouses, shipping zones, and channels must be properly linked together
+to provide the expected allocation. Below you can find the explanation of the terms
+that will be required to properly understand this guide:
+
+- channel - allow configuring your shop differently for different currencies;
+  go [here](./channels.mdx) for more information
+- shipping zone - allow for defining the cost of shipping in different areas of the world,
+  shipping zone contains shipping methods that can be used for a given part of the world
+- warehouse - contains stocks of specific products, assigned to a given channel and shipping zone,
+  can be defined as the place for click and collect
+- stock - corresponds to resources of specific product variant in the given warehouse
+- allocation - corresponds to the currently allocated stocks for a given order line
+
+The relations between the mentioned entities are shown in the below chart:
+
+```mermaid
+classDiagram
+    Channel "*" -- "*" ShippingZone
+    ShippingZone "1" -- "*" ShippingMethod
+    Channel "*" -- "*" Warehouse
+    Warehouse "*" -- "*" ShippingZone
+    Warehouse "1" -- "*" Stock
+    Stock "*" -- "1" ProductVariant
+    Stock "1" -- "*" Allocation
+    Allocation "*" -- "1" OrderLine
+    class Stock{
+      quantity: Number
+      quantity_allocated: Number
+    }
+    class Allocation{
+      quantity_allocated: Number
+    }
+    class Warehouse{
+      click_and_collect_option: String
+    }
+```
+
+## Allocation flow
+
+Stocks are allocated in the checkout completing process, for all lines
+that have track inventory turned on. The allocation is made for every line
+on stocks with the highest quantity available value. When there is not enough
```

Comment:
```suggestion
that have track inventory turned on. The allocation is made for every line
on stocks with the highest quantity value available. When there is not enough
```

I can be wrong, though, because I am not sure if I understand the context correctly.

Reply:
IMO it shouldn't be changed. I'm writing about the value that is called `available quantity` and it's the same as the `quantity`. Some `quantity` must be allocated or reserved and in such case, both values differs.

Reply:
Got it! Thanks for explanation.